docs/licensing-policy.md

```diff
@@ -0,0 +1,47 @@
+# IPFS Licensing Policy
+
+In order to ensure an open source project is actually open and re-usable in a legal sense, it’s critical that licensing information is clear. Repositories in the `ipfs`, `libp2p`, and `ipld` organizations should all follow this policy for licensing their contents.
+
+Unless there is a particular reason to use a different licensing structure, please use the following for new projects:
+
+1. Copyright should always be assigned to **Protocol Labs, Inc,** regardless of the type of license. (In an ideal future, we can assign copyright to a foundation, but Protocol Labs is the best usable legal entity we have now.)
+
+2. All software code should be licensed under the MIT license. Canonical license text for the MIT license can be found in this repository’s [`LICENSE` file](../LICENSE) or at the [Open Source Initiative website](https://opensource.org/licenses/MIT).
+
+2. All documentation and non-code resources (e.g. logos and imagery) should be licensed under the Creative Commons Attribution-ShareAlike 4.0 International (CC BY-SA 4.0) license. An [easy-to-read, linkable description](https://creativecommons.org/licenses/by-sa/4.0/) and canonical license text in [HTML](https://creativecommons.org/licenses/by-sa/4.0/legalcode) and [plain text](https://creativecommons.org/licenses/by-sa/4.0/legalcode.txt) can be found at the Creative Commons website.
+
+    If a repository is a software project with some built-in docs or resources (e.g. `go-ipfs`), it can all be licensed under the MIT license. The CC BY-SA license is meant for projects that are not overwhelmingly code, such as websites, guidance, discussion repos, and so on.
+
+
+## Attaching a License
```

Comment:
Just an idea: it may be super helpful if we had some form of license validation/automation that could be dropped into existing release pipelines to check if things mentioned in this section are valid on every build.

Most of our projects is in JS and Golang. [aegir](https://github.com/ipfs/aegir) could provide script for that out-of-the-box, something similar could be created for Golang as well.

Reply:
That could definitely be useful, but I think it might be a little out-of-scope here:

- We need to agree on a clear policy before we can validate that something fits the policy :)
- There are at least a few open questions that would need to be answered to build that tool. I think we should be careful not to let that derail just agreeing on a policy:
    - There will always need to be some exceptions to that policy, so you’ve got to figure out how any given tool will handle that.
    - What’s validatable here? The suggested README language is intentionally only a suggestion; there are lots of other good ways to word it.
    - What about repos that don’t have a build process? (e.g. `ipfs/ipfs`)
    - What about https://project-repos.ipfs.io ?

Maybe this particular discussion is better in a separate issue?